pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -417,13 +430,17 @@ func (r *ReconcileClusterSync) Reconcile(ctx context.Context, request reconcile.
 		ownerRef := metav1.NewControllerRef(cd, cd.GroupVersionKind())
 		ownerRef.Controller = nil
 		clusterSync.OwnerReferences = []metav1.OwnerReference{*ownerRef}
-		if err := r.Create(context.Background(), clusterSync); err != nil {
+		switch err := r.Create(context.Background(), clusterSync); {
+		case apierrors.IsAlreadyExists(err):
+			// race condition, just proceed
+		case err != nil:
 			logger.WithError(err).Log(controllerutils.LogLevel(err), "could not create ClusterSync")
 			return reconcile.Result{}, err
+		default: // clusterSync was created successfully
+			recobsrv.SetOutcome(hivemetrics.ReconcileOutcomeClusterSyncCreated)
+			// requeue immediately so that we reconcile soon after the ClusterSync is created
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
This explicit requeue _should_ be redundant.
```suggestion
			// requeue immediately so that we reconcile soon after the ClusterSync is created
			return reconcile.Result{}, nil
```
but I'm good with leaving it here.